test/integration/src/test/java/io/pravega/test/integration/ControllerRestApiTest.java

```diff
@@ -191,6 +205,29 @@ public void restApiTests() {
         Assert.assertEquals("List streams size", 1, response.readEntity(StreamsList.class).getStreams().size());
         log.info("List streams successful");
 
+        // Test listStream GET /v1/scopes/scope1/streams for tags
+        response = client.target(resourceURl).queryParam("filter_type", "tag").
+                queryParam("filter_value", "testTag").request().get();
+        assertEquals("List streams", OK.getStatusCode(), response.getStatus());
+        Assert.assertEquals("List streams size", 1, response.readEntity(StreamsList.class).getStreams().size());
+
+        response = client.target(resourceURl).queryParam("filter_type", "tag").
+                queryParam("filter_value", "randomTag").request().get();
+        assertEquals("List streams", OK.getStatusCode(), response.getStatus());
+        Assert.assertEquals("List streams size", 0, response.readEntity(StreamsList.class).getStreams().size());
+        log.info("List streams with tag successful");
+
+        response = client.target(resourceURl).queryParam("filter_type", "showInternalStreams").request().get();
+        assertEquals("List streams", OK.getStatusCode(), response.getStatus());
+        assertTrue(response.readEntity(StreamsList.class).getStreams().get(0).getStreamName().startsWith("_MARK"));
```

Comment:
Here we're just testing the -ve test case where Scope does not have any internal Streams under it
Please also add a positive test case where the scope does have atleast 1 internal Streams under it.

Reply:
The code highlighted in the comment is for the positive test case for an internal stream. I use the `startsWith` String method to check if the stream is internal.
The negative case is at line 229. Uses the `listStreams` call without filters to see that an empty list is returned.